### ISSUE TITLE
systemd user service file

### ISSUE THREAD

**Issue by dsalt** [dsalt/devilspie2#2]
https://bugs.debian.org/933866

>It would be wonderful to have Debian ship a systemd user service file for
devilspie2. I've searched the internet a bit and know a bit about systemd, but
don't know the exact configurations for said service file. My current attempts
are failing. If I get it to work, I'll follow-up to this bug report.
>
>It'd be great to be able to:
>
>systemctl --user enable devilspie2
>systemctl --user start devilspie2
>
>and have it work out of the box.

Being a Devuan user, I don't use systemd myself, so I'm in no position to add this. (I currently use Xfce4's application autostart to run devilspie2 on login.)

**Comment by Longoon12000** [comment]
I don't think that using systemd is the correct way to run devilspie2, the main issue being that it should run when you **login** to your window manager which is not a systemd target. It should have a seperate instance for each session.
The corresponding WM's application autostart is the place that make sense to start it with.

https://unix.stackexchange.com/a/360685

**Comment by cwh-0** [comment]
*Disclaimer: I don't know much systemd*

Could extend this feature request to be more implementation-agnostic - "run devilspie2 automatically", to consider other solutions like the autostart suggestion.

---

@Longoon12000 I would think this has got to be a common enough condition for there to be a wm-agnostic set of unit constraints in systemd:  "user has logged into an X session". It doesn't have to be just a single systemd `.target` that expresses this. Also, in your link:

> https://unix.stackexchange.com/a/360685

this example seems to be installed as a root entry (`/etc/systemd/system/`), not a user entry, per the original request (`systemctl --user`). I think that changes things. From the README, it sounds like devilspie2 is intended to be run per-user:

> If you don't give devilspie2 any folder with --folder, it will read Lua scripts from the folder that it gets from the GLib function g_get_user_config_dir with devilspie2/ added to the end.

So what about a _user_ service `After=graphical.target` installed to `/etc/systemd/user`?

---

That said, I did this:
* for gnome
* as the only person using the machine
* never accessing remotely e.g. ssh

`$HOME/.local/systemd/user/devilspie2.service`
```
[Service]
Type=simple
ExecStart=/usr/local/bin/devilspie2 --debug
Restart=on-failure

[Install]
WantedBy=gnome-session-x11-services.target
```

I could play around with a more generic solution.

**Comment by dsalt** [comment]
Valid point re. desktop/wm autostart (so, presumably, XDG autostart). I've just pushed a commit which _should_ allow for that.
